docs/user/troubleshooting.md

```diff
@@ -0,0 +1,87 @@
+# Installer Troubleshooting
+
+Unfortunately, there will always be some cases where OpenShift fails to install properly. In these events, it is helpful to understand the likely failure modes as well as how to troubleshoot the failure.
+
+## Common Failures
+
+### No Workers Nodes Created
+
+The installer doesn't provision worker nodes directly, like it does with master nodes. Instead, the cluster relies on the Machine API Operator, which is able to scale up and down nodes on supported platforms. If more than fifteen to twenty minutes (depending on the speed of the cluster's Internet connection) have elapsed without any workers, the Machine API Operator needs to be investigated.
+
+The status of the Machine API Operator can be checked by running the following command from the machine used to install the cluster:
+
+```
+oc --config=${INSTALL_DIR}/auth/kubeconfig --namespace=openshift-cluster-api get pods
+```
+
+If the API is unavailable, that will need to be [investigated first](#api-is-unavailable).
+
+The previous command should yield output similar to the following:
+
+```
+NAME                                             READY     STATUS    RESTARTS   AGE
+clusterapi-manager-controllers-774dc4557-nx5xq   3/3       Running   0          4h
+machine-api-operator-7894d8f85-lq2ts             1/1       Running   0          4h
+```
+
+The logs for the machine-controller container within the `clusterapi-manager-controllers` pod need to be checked to determine why the workers haven't been created. That can be done with the following (the exact name of the pod will need to be substituted):
+
+```
+oc --config=${INSTALL_DIR}/auth/kubeconfig --namespace=openshift-cluster-api logs clusterapi-manager-controllers-774dc4557-nx5xq --container=machine-controller
+```
+
+### API is Unavailable
+
+When the API is unavailable, the master nodes will need to checked to ensure that they are running the correct components. This requires SSH access so it is necessary to include an administrator's SSH key during the installation.
+
+If SSH access to the master nodes isn't available, that will need to be [investigated next](#unable-to-ssh-into-master-node).
+
+The first thing to check is to make sure that etcd is running on each of the masters. The etcd logs can be viewed by running the following on each master node:
+
+```
+sudo crictl logs $(sudo crictl ps --pod=$(sudo crictl pods --name=etcd-member --quiet) --quiet)
+```
+
+If the previous command fails, ensure that the etcd pods have been created by the Kubelet:
+
+```
+sudo crictl pods --name=etcd-member
+```
+
+If no pods are shown, etcd will need to be [investigated](#etcd-is-not-running).
+
+### Unable to SSH into Master Nodes
+
+In order to SSH into the master nodes, it is necessary to include an administrator's SSH key during the installation. If the nodes are accessible over the network, but SSH authentication is failing, ensure that the proper SSH key is being used.
```

Comment:
"accessible over the network" isn't very specific.  Are you talking about the Kubernetes API?  Ping responses?  Something else?  In the past, we've had issues where folks set broken libvirt images (e.g. I forgot to unzip one) which left the nodes unbootable.  The nodes existed, but you had to grab their console to see the hang.  Is it worth adding a console-grabbing section (I see you mention it below, but without explicit command suggestions for AWS or libvirt)?  When that does work, Ignition will talk about the keys it's setting, which also helps debug SSH access.